### DISCUSSION TITLE
Are you guaranteed to get the same order back when using Prefer: return=representation?

### DISCUSSION THREAD

**Discussion by ewan-escience** [PostgREST/postgrest#2953] [Q&A] [answered]
When you insert/upsert multiple rows in a table and use `Prefer: return=representation`, is the data you get back guaranteed to be in the same order as the array in which you submitted the data?

**Comment by wolfgangwalther** [comment, marked as answer]
No, there is no such guarantee. See also a thread on pgsql-general from earlier this year: https://www.postgresql.org/message-id/flat/2b8d529a-0ec-4e8f-26d3-2ef2881a2d49%40evolvis.org#3fdbbeb01a034274951b238c30854704

Given that there is no "proper" way to do it in PostgreSQL itself, we don't do it here either.

**Reply by ewan-escience** [reply]
That's unfortunate, but thanks for the quick answer!
